pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -489,15 +490,23 @@ func (jm *ControllerV2) syncCronJob(
 		// this is likely a user error in defining the spec value
 		// we should log the error and not reconcile this cronjob until an update to spec
 		klog.V(2).InfoS("Unparseable schedule", "cronjob", klog.KRef(cj.GetNamespace(), cj.GetName()), "schedule", cj.Spec.Schedule, "err", err)
-		jm.recorder.Eventf(cj, corev1.EventTypeWarning, "UnparseableSchedule", "unparseable schedule: %s : %s", cj.Spec.Schedule, err)
+		jm.recorder.Eventf(cj, corev1.EventTypeWarning, "UnparseableSchedule", "unparseable schedule: %q : %s", cj.Spec.Schedule, err)
 		return cj, nil, nil
 	}
+
+	if strings.Contains(cj.Spec.Schedule, "CRON_TZ") {
+		jm.recorder.Eventf(cj, corev1.EventTypeWarning, "UnsupportedSchedule", "CRON_TZ used in schedule %q is not officially supported, see https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/ for more details", cj.Spec.Schedule)
+	}
+	if strings.Contains(cj.Spec.Schedule, "TZ") {
```

Comment:
the event will hit twice if we have `CRON_TZ`. Same in `strategy.go`

Reply:
True, lemme fix that.

Reply:
Updated.